tests/ld-key-pair.spec.js

```diff
@@ -22,6 +23,28 @@ describe('LDKeyPair', () => {
   describe('Ed25519KeyPair', () => {
     const type = 'Ed25519VerificationKey2018';
 
+    describe('constructor', () => {
+      it('should auto-set key.id based on controller, if present', async () => {
+        const {publicKeyBase58} = mockKey;
+        const controller = 'did:example:1234';
+
+        const keyPair = new Ed25519KeyPair({controller, publicKeyBase58});
+        expect(keyPair.id).to.equal(
+          'did:example:1234#z6Mks8wJbzhWdmkQZgw7z2qHwaxPVnFsFmEZSXzGkLkvhMvL');
+      });
+
+      it('should error if publicKeyBase58 property is missing', async () => {
+        let error;
+        try {
+          new Ed25519KeyPair({});
+        } catch(e) {
+          error = e;
+        }
+        expect(error.message)
```

Comment:
Would asserting that the error is an instanceOf `TypeError` have caught the missing new?

Reply:
Unfortunately, no. (I added that assertion, just in case, tho.)
The problem is, for whatever reason, `throw TypeError('whatever')` legitimately throws that exact type error. So, there's no way to detect that lack of new I think.